src/modules/hip/kernel/vignette.hpp

```diff
@@ -0,0 +1,320 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ void vignette_gaussian_hip_compute(float &multiplier, int2 &halfDimsWH_i2, int2 &idXY_i2, d_float8 *gaussianValue_f8)
+{
+    float rowLocComponent;
+    rowLocComponent = idXY_i2.y - halfDimsWH_i2.y;
+    rowLocComponent *= (rowLocComponent);
+
+    float4 rowLocComponent_f4 = static_cast<float4>(rowLocComponent);
+    float4 multiplier_f4 = static_cast<float4>(multiplier);
+
+    d_float8 colLocComponent_f8;
+    colLocComponent_f8.f4[0] = make_float4(idXY_i2.x, idXY_i2.x + 1, idXY_i2.x + 2, idXY_i2.x + 3);
+    colLocComponent_f8.f4[1] = colLocComponent_f8.f4[0] + static_cast<float4>(4);
+    colLocComponent_f8.f4[0] -= static_cast<float4>(halfDimsWH_i2.x);
+    colLocComponent_f8.f4[1] -= static_cast<float4>(halfDimsWH_i2.x);
+    colLocComponent_f8.f4[0] = (colLocComponent_f8.f4[0] * colLocComponent_f8.f4[0]) + rowLocComponent_f4;
+    colLocComponent_f8.f4[1] = (colLocComponent_f8.f4[1] * colLocComponent_f8.f4[1]) + rowLocComponent_f4;
+    colLocComponent_f8.f4[0] = colLocComponent_f8.f4[0] * multiplier_f4;
+    colLocComponent_f8.f4[1] = colLocComponent_f8.f4[1] * multiplier_f4;
+
+    gaussianValue_f8->f4[0] = make_float4(expf(colLocComponent_f8.f4[0].x), expf(colLocComponent_f8.f4[0].y), expf(colLocComponent_f8.f4[0].z), expf(colLocComponent_f8.f4[0].w));
+    gaussianValue_f8->f4[1] = make_float4(expf(colLocComponent_f8.f4[1].x), expf(colLocComponent_f8.f4[1].y), expf(colLocComponent_f8.f4[1].z), expf(colLocComponent_f8.f4[1].w));
+}
+
+__device__ void vignette_8_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, d_float8 *gaussianValue_f8)
+{
+    dst_f8->f4[0] = src_f8->f4[0] * gaussianValue_f8->f4[0];
+    dst_f8->f4[1] = src_f8->f4[1] * gaussianValue_f8->f4[1];
+    dst_f8->f1[0] = nearbyintf(dst_f8->f1[0]);
+    dst_f8->f1[1] = nearbyintf(dst_f8->f1[1]);
+    dst_f8->f1[2] = nearbyintf(dst_f8->f1[2]);
```

Comment:
Pls create a device helper under the hip_math_* group for this nearbyintf. It should take in a *src_f8 and return a *dst_f8. For this use case, you can pass in dst_f8 for both arguments from here.